automl/automl-library/library.whizzml

```diff
@@ -242,29 +266,33 @@
           (lambda (i) (_flatline-item (items i) (fields ((items i) "field_id"))))
           lhs (map get-item-fn (rule "lhs"))
           rhs (map get-item-fn (rule "rhs"))
+          metric (rule "leverage" 1)
           lhs (if (> (count lhs) 1)
                 (str "(and " (join " " lhs) ")")
                 (join " " lhs))
           rhs (if (> (count rhs) 1)
                 (str "(and " (join " " rhs) ")")
                 (join " " rhs)))
-      (cond (= match "antecedent")
-            lhs
-            (= match "consequent")
-            rhs
-            (str "(and " lhs " " rhs ")")))))
-
-(define (_batch-association-sets dataset-id association-id  match . rule-ids)
+      (when (>= (abs metric) threshold)
+        (cond (= match "antecedent")
+              lhs
+              (= match "consequent")
+              rhs
+              (str "(and " lhs " " rhs ")"))))))
+
+(define (_batch-association-sets dataset-id association-id match threshold)
   (let (association (fetch association-id)
         associations (association "associations" {})
         rules (associations "rules" [])
-        rule-ids (if (empty? (head rule-ids []))
-                   (map (lambda (r) (r "id")) rules)
-                   (head rule-ids))
+        rule-ids (map (lambda (r) (r "id")) rules)
```

Comment:
If you don't filter by the list of rule-ids provided as parameter in the function, then these two following sentences are useless, right?

Reply:
Yes, I didn't notice!